jmeter: add page (#8391)
tldr-pages/tldr: pages/common/jmeter.md

# jmeter

> Open source java application designed for load testing functional behavior and measure performance.
> More information: <https://jmeter.apache.org>.

- Run in nongui mode for a specific test plan:

> `jmeter --nongui --testfile {{path/to/file}}.jmx`
> - Use a specific log file:

Reviewer: ```suggestion
`jmeter --nongui --testfile {{path/to/file}}.jmx`

- Use a specific log file:
```

# jmeter

> Open source java application designed for load testing functional behavior and measure performance.

Reviewer 2: ```suggestion
> Open source java application designed for load testing functional behavior and measure performance.
```

Author: Done, please validate! ✅

> > Open source java application designed for load testing functional behavior and measure performance.

Reviewer 2: I have a suggestion should this line be 
```suggestion
> Open source java application designed for load testing functional behavior and measuring performance.
```
 or 
```suggestion
> Open source java application designed for load testing functional behavior and to measure performance.
 ```
 I think the current one feels a bit odd with *measure perforamance* alone. What do you guys think @navarroaxel @CleanMachine1 @EmilySeville7cfg ?

Reviewer 3: I prefer leave as-is.

> More information: <https://jmeter.apache.org/>.

Reviewer 2: ```suggestion
> More information: <https://jmeter.apache.org>.
```
I don't think the last slash is necessary.

Author: Done, please validate! ✅

> - Run jmeter in cli mode for a test plan:

Reviewer: ```suggestion
- Run in [n]ongui mode for a specific [t]est plan:
```

Author: Makes sense, please validate ✅

> - Run in nongui mode for a specific test plan:

Reviewer 3: Perhaps this better
```suggestion
- Run a specific test plan in nongui mode:
```

> `jmeter -n -t {{name}}.jmx`

Reviewer: ```suggestion
`jmeter --nongui --testfile {{path/to/file}}.jmx`
```

```suggestion
`jmeter --nongui --testfile {{path/to/file.jmx}}`
```

Is `.jmx` a required extension? Or it can be any? Choose the second suggestion if extension is required but it doesn't matter what it is.

Author: It is a required extension for jmeter to boot up in this case. 
Please validate! ✅

>     

Reviewer 4: ```suggestion

```

> - Run jmeter and specify the log output:

Reviewer: ```suggestion
- Specify a specific log file:
```

Author: ✅

> - Use a specific log file:

Reviewer 3: Isolated examples are best, and self explanatory
```suggestion
- Run a test plan in nongui mode using a specific log file:
```

> `jmeter -n -t {{name}}.jmx -l {{logfile}}.jtl`

Reviewer: - We prefer long options. There is no fixed convention in our style guide when we do this. But it will be added soon.
- Don't use plain `name` or `logfile` as above. They are paths here so they must be written as:

```suggestion
`jmeter --nogui --testfile {{path/to/file}}.jmx --logfile {{path/to/logfile}}.jtl`
```

```suggestion
`jmeter --nogui --testfile {{path/to/file.jmx}} -l {{path/to/logfile.jtl}}`
```

Author: Yep, looks more readable. 
Please validate ✅

> - Run jmeter using a proxy server and a port:

Reviewer: ```suggestion
- Use a specific proxy server and a specific port:
```

Author: ✅

> `jmeter -n -t {{name}}.jmx -H {{proxy_server_url}} -P {{port}}`

Reviewer: It's better to provide a sample value instead of `proxy_server_url`:

```suggestion
`jmeter --nongui --testfile {{path/to/file}}.jmx --proxyHost {{proxy_server_url}} --proxyPort {{port}}`
```

```suggestion
`jmeter --nongui --testfile {{path/to/file.jmx}} --proxyHost {{proxy_server_url}} --proxyPort {{port}}`
```

Author: I was not too sure if I should specify or not.
Please validate ✅

> - Use a specific proxy server and a specific port:

Reviewer 3: I think port and host are implicit in "proxy" 
```suggestion
- Run a test plan in nongui mode using a specific proxy:
```

> - Run jmeter with user defined variables:

Reviewer: ```suggestion
- Define user variables:
```

> `jmeter -J{{key}}={{value}} -n -t {{file_name}}.jmx`

Reviewer: ```suggestion
`jmeter --jmeterproperty {{key}}='{{value}}' --nongui --testfile {{path/to/file}}.jmx`
```

```suggestion
`jmeter --jmeterproperty {{key}}='{{value}}' --nongui --testfile {{path/to/file.jmx}}`
```

Author: ✅

> - Define user variables:

Reviewer 3: The man page <https://manned.org/jmeter> refers to this variables as `JMeter properties` so:
```suggestion
- Run a test plan in nongui mode using a specific JMeter property:
```
